src/java/org/apache/cassandra/auth/CIDRPermissionsCache.java

```diff
@@ -54,7 +54,7 @@ public CIDRPermissionsCache(Function<RoleResource, CIDRPermissions> loadFunction
      */
     public boolean invalidateCidrPermissions(String roleName)
     {
-        if (cache.getIfPresent(roleName) == null)
+        if (cache.getIfPresent(RoleResource.role(roleName)) == null)
```

Comment:
Thanks for the fix! It looks good. 
The original code does not actually invalidate from cache. (And I wrote a test that verifies it)

Reply:
Thanks @yifan-c. There is actually this test class - InvalidateCIDRPermissionsCacheTest but it doesn't seem to test properly. It completes successfully with and without the change made. I think it doesn't really assert that invalidation happens. What test did you write?

Reply:
yeah. Specifically, I changed the test, `testInvalidateSingleCidrPermission`. The assertions before and after invalidation were missing. The updated test fails before your patch. 

```java
        assertThat(((CassandraCIDRAuthorizer) DatabaseDescriptor.getCIDRAuthorizer()).has(ROLE_A)).isTrue();

        // invalidate cidr permission
        ToolRunner.ToolResult tool = ToolRunner.invokeNodetool("invalidatecidrpermissionscache", ROLE_A.getRoleName());
        tool.assertOnCleanExit();

        assertThat(((CassandraCIDRAuthorizer) DatabaseDescriptor.getCIDRAuthorizer()).has(ROLE_A)).isFalse();
```

Reply:
Hmmm, there is no such method "has" :-) And I do not think you meant "hasAccessFromIp".

Reply:
I added a `has` method for testing...

Reply:
Updated the tests and added small condition in the nodetool command so we are sure we do not regress again in the future. Can you, please, take a look?

Reply:
```suggestion
        if (!cache.asMap().containsKey(RoleResource.role(roleName)))
```

Reply:
That's awesome, thank you @ben-manes ! I appreciate your feedback. Do you mind checking if there might be anything else I might have missed with this upgrade?

Reply:
lgtm, thanks for upgrading!

Reply:
please create `RoleResource.role` only once